protocol/src/protocol.semanticHighlighting.proposed.ts

```diff
@@ -0,0 +1,91 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) TypeFox. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import { NotificationType } from 'vscode-jsonrpc';
+import { VersionedTextDocumentIdentifier } from 'vscode-languageserver-types';
+
+/**
+ * Parameters for the semantic highlighting (server-side) push notification.
+ */
+export interface SemanticHighlightingParams {
+
+	/**
+	 * The text document that has to be decorated with the semantic highlighting information.
+	 */
+	textDocument: VersionedTextDocumentIdentifier;
+
+	/**
+	 * An array of semantic highlighting information.
+	 */
+	lines: SemanticHighlightingInformation[];
+
+}
+
+/**
+ * Represents a semantic highlighting information that has to be applied on a specific line of the text document.
+ */
+export interface SemanticHighlightingInformation {
+
+	/**
+	 * The zero-based line position in the text document.
+	 */
+	line: number;
+
+	/**
+	 * A base64 encoded string representing every single highlighted characters with its start position, length and the "lookup table" index of
```

Comment:
Which type of base64 encoding is used here? (standard, url_safe? padding?) (https://docs.rs/base64/0.11.0/base64/index.html#constants) cc https://github.com/gluon-lang/lsp-types/pull/127